pkg/xds/envoy/routes/tags_configurer.go

```diff
@@ -0,0 +1,32 @@
+package routes
+
+import (
+	envoy_api_v2 "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	envoy_core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+
+	"github.com/Kong/kuma/api/mesh/v1alpha1"
+)
+
+const TagsHeader = "x-kuma-tags"
+
+func Tags(tags v1alpha1.MultiValueTagSet) RouteConfigurationBuilderOpt {
+	return RouteConfigurationBuilderOptFunc(func(config *RouteConfigurationBuilderConfig) {
+		config.Add(&TagsConfigurer{
+			tags: tags,
+		})
+	})
+}
+
+type TagsConfigurer struct {
+	tags v1alpha1.MultiValueTagSet
+}
+
+func (t *TagsConfigurer) Configure(rc *envoy_api_v2.RouteConfiguration) error {
+	if len(t.tags) == 0 {
+		return nil
+	}
+	rc.RequestHeadersToAdd = append(rc.RequestHeadersToAdd, &envoy_core.HeaderValueOption{
+		Header: &envoy_core.HeaderValue{Key: TagsHeader, Value: t.tags.String()},
```

Comment:
1) I would not tie functionality to `.String()` function. `.String()` is meant for debugging and _maybe_ for displaying information.
2) I don't believe this format `tag1=value11,value12 tag2=value21 tag3=value31,value32,value33` will be right for the regex. I think we wanted something like this `&tag1=value11,value12&tag2=value21&tag3=value31,value32,value33&` 

Also, regex has to support multi-values.
```
sources:
  version: 1
```
should match `version=1,2`

Additionally, we need to add validation for tag value (can be in a separate PR) that prevents user for using special symbols.

Reply:
Sure, I'll create separate method.
There is no problem with regex and multi-values:
`&version=[^&]*1.*` will match the string `version=1,2`

Reply:
btw string with ampersand should be quoted in order yaml to be valid, maybe whitespace is good delimiter?

Reply:
It's not like it's our job to do it - proto -> yaml serializer will handle this. I'd stick it to `&`. It's a matter of what character users won't be able to use in the tags value field.